ui/components/graphql/subscriptions/ServiceMeshSubscription.js

```diff
@@ -4,23 +4,34 @@ import { requestSubscription } from "react-relay";
 import environment from "../../../lib/relayEnvironment";
 
 export default function subscribeServiceMeshEvents(dataCB, variables) {
+  let controlRes = null
+
   requestSubscription(environment, {
-    subscription : dataPlaneSubscription,
+    subscription : controlPlaneSubscription,
     variables : {
       filter : variables,
     },
-    onNext : (controlPlanesRes) => {
-      dataCB(controlPlanesRes, null)
-      requestSubscription(environment, {
-        subscription : controlPlaneSubscription,
-        variables : {
-          filter : variables,
-        },
-        onNext : (dataPlanesRes) => dataCB(controlPlanesRes, dataPlanesRes),
-        onError : (error) => console.log(`An error occured:`, error),
-      });
+    onNext : (res) => {
+      // decrease state change in dashboard
+
+      // dataCB(res, null)
+      controlRes = res
     },
     onError : (error) => console.log(`An error occured:`, error),
   });
 
+  requestSubscription(environment, {
+    subscription : dataPlaneSubscription,
+    variables : {
+      filter : variables,
+    },
+    onNext : (dataRes) => {
+      if (controlRes) dataCB(controlRes, dataRes)
+    },
+    onError : (error) => {
+      if (controlRes) dataCB(controlRes, null)
+      console.log(`An error occured:`, error)
```

Comment:
Let’s examine what we might want to let the user know in order to have them troubleshoot the situation.